MBBSEmu.Tests/CPU/FSTP_Tests.cs

```diff
@@ -0,0 +1,77 @@
+﻿using Iced.Intel;
+using System;
+using Xunit;
+using static Iced.Intel.AssemblerRegisters;
+
+namespace MBBSEmu.Tests.CPU
+{
+    public class FSTP_Tests : CpuTestBase
+    {
+        [Theory]
+        [InlineData(double.MaxValue)]
+        [InlineData(double.MinValue)]
+        [InlineData(0.0d)]
+        [InlineData(double.NaN)]
+        public void FSTP_Test_ST1(double ST0Value)
+        {
+            Reset();
+            mbbsEmuCpuRegisters.Fpu.SetStackTop(1);
+            mbbsEmuCpuCore.FpuStack[1] = ST0Value;
+
+            var instructions = new Assembler(16);
+            instructions.fstp(st1);
```

Comment:
is there an assert that ST1 == ST0Value?

Reply:
In this test, ST1 becomes ST0 after the FSTP instruction because of the FPU POP. So we set the FPU Stack Top to [1], FSTP saves the result to ST1 [0] and the pop sets ST0 to [0].